### ISSUE TITLE
Not working with 4090

### ISSUE THREAD

**Issue by kdcyberdude** [huggingface/instruction-tuned-sd#6]
Hi @sayakpaul, Thanks for writing such a fantastic blog.

I am having difficulty making it work with 4090. The loss is converged to NaN. 
The only change that I have made because of the VRAM constraint is using `--use_8bit_adam` and removed `--enable_xformers_memory_efficient_attention ` since I am using the latest version of PyTorch and cuda(12.1).

**Logs -** 

> export MODEL_ID="timbrooks/instruct-pix2pix"
> export DATASET_ID="instruction-tuning-sd/cartoonization"
> export OUTPUT_DIR="cartoonization-finetuned"
> 
> accelerate launch --mixed_precision="fp16" finetune_instruct_pix2pix.py \
>   --pretrained_model_name_or_path=$MODEL_ID \
>   --dataset_name=$DATASET_ID \
>   --use_ema \
>   --use_8bit_adam \
>   --resolution=256 --random_flip \
>   --train_batch_size=2 --gradient_accumulation_steps=4 --gradient_checkpointing \
>   --max_train_steps=15000 \
>   --checkpointing_steps=5000 --checkpoints_total_limit=1 \
>   --learning_rate=5e-05 --lr_warmup_steps=0 \
>   --mixed_precision=fp16 \
>   --val_image_url="https://hf.co/datasets/diffusers/diffusers-images-docs/resolve/main/mountain.png" \
>   --validation_prompt="Generate a cartoonized version of the natural image" \
>   --seed=42 \
>   --output_dir=$OUTPUT_DIR \
>   --report_to=wandb \
>   --push_to_hub

```
The following values were not passed to `accelerate launch` and had defaults used instead:
	`--num_processes` was set to a value of `1`
	`--num_machines` was set to a value of `1`
	`--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
WARNING[XFORMERS]: xFormers can't load C++/CUDA extensions. xFormers was built for:
    PyTorch 2.0.1+cu118 with CUDA 1108 (you have 2.1.0.dev20230528)
    Python  3.11.3 (you have 3.11.3)
  Please reinstall xformers (see https://github.com/facebookresearch/xformers#installing-xformers)
  Memory-efficient attention, SwiGLU, sparse and more won't be available.
  Set XFORMERS_MORE_DETAILS=1 for more details
/home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/accelerate/accelerator.py:258: FutureWarning: `logging_dir` is deprecated and will be removed in version 0.18.0 of 🤗 Accelerate. Use `project_dir` instead.
  warnings.warn(
05/29/2023 15:04:28 - INFO - __main__ - Distributed environment: DistributedType.NO
Num processes: 1
Process index: 0
Local process index: 0
Device: cuda

Mixed precision type: fp16

Cloning https://huggingface.co/kdcyberdude/cartoonization-finetuned into local empty directory.
05/29/2023 15:04:32 - WARNING - huggingface_hub.repository - Cloning https://huggingface.co/kdcyberdude/cartoonization-finetuned into local empty directory.
{'variance_type', 'clip_sample_range'} was not found in config. Values will be initialized to default values.
{'scaling_factor'} was not found in config. Values will be initialized to default values.
{'class_embeddings_concat', 'conv_in_kernel', 'encoder_hid_dim', 'time_embedding_dim', 'resnet_out_scale_factor', 'timestep_post_act', 'time_embedding_act_fn', 'conv_out_kernel', 'projection_class_embeddings_input_dim', 'addition_embed_type_num_heads', 'cross_attention_norm', 'time_embedding_type', 'resnet_skip_time_act', 'addition_embed_type', 'time_cond_proj_dim', 'mid_block_only_cross_attention'} was not found in config. Values will be initialized to default values.

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please run

python -m bitsandbytes

 and submit this information together with your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
================================================================================
bin /home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda121.so
/home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/bitsandbytes/cuda_setup/main.py:149: UserWarning: /home/kd/anaconda3/envs/sd did not contain ['libcudart.so', 'libcudart.so.11.0', 'libcudart.so.12.0'] as expected! Searching further paths...
  warn(msg)
CUDA SETUP: CUDA runtime path found: /usr/local/cuda-12/lib64/libcudart.so
CUDA SETUP: Highest compute capability among GPUs detected: 8.9
CUDA SETUP: Detected CUDA version 121
CUDA SETUP: Loading binary /home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda121.so...
Downloading readme: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.31k/1.31k [00:00<00:00, 2.56MB/s]
Downloading and preparing dataset None/None to /home/kd/.cache/huggingface/datasets/instruction-tuning-sd___parquet/instruction-tuning-sd--cartoonization-671b5867b296e9e9/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec...
Downloading data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 472M/472M [00:26<00:00, 18.0MB/s]
Downloading data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 443M/443M [00:25<00:00, 17.6MB/s]
Downloading data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 447M/447M [00:24<00:00, 18.1MB/s]
Downloading data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 474M/474M [00:26<00:00, 17.7MB/s]
Downloading data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 467M/467M [00:26<00:00, 17.4MB/s]
Downloading data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 484M/484M [00:37<00:00, 12.8MB/s]
Downloading data: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 510M/510M [00:28<00:00, 17.8MB/s]
Downloading data files: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [03:37<00:00, 217.10s/it]
Extracting data files: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 874.18it/s]
Dataset parquet downloaded and prepared to /home/kd/.cache/huggingface/datasets/instruction-tuning-sd___parquet/instruction-tuning-sd--cartoonization-671b5867b296e9e9/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec. Subsequent calls will reuse this data.     
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 51.75it/s]
wandb: Currently logged in as: kdcyberdude. Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.15.3
wandb: Run data is saved locally in /home/kd/Desktop/proj/ai/hf/instruction-tuned-sd/wandb/run-20230529_150832-kflr0nm1
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run decent-eon-3
wandb: ⭐️ View project at https://wandb.ai/kdcyberdude/instruct-pix2pix-cartoonizer
wandb: 🚀 View run at https://wandb.ai/kdcyberdude/instruct-pix2pix-cartoonizer/runs/kflr0nm1
05/29/2023 15:08:35 - INFO - __main__ - ***** Running training *****
05/29/2023 15:08:35 - INFO - __main__ -   Num examples = 5000
05/29/2023 15:08:35 - INFO - __main__ -   Num Epochs = 24
05/29/2023 15:08:35 - INFO - __main__ -   Instantaneous batch size per device = 2
05/29/2023 15:08:35 - INFO - __main__ -   Total train batch size (w. parallel, distributed & accumulation) = 8
05/29/2023 15:08:35 - INFO - __main__ -   Gradient Accumulation steps = 4
05/29/2023 15:08:35 - INFO - __main__ -   Total optimization steps = 15000
Steps:   0%|                                                                                                                                                                                                                                                    | 0/15000 [00:00<?, ?it/s]/home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/torch/nn/modules/conv.py:459: UserWarning: Applied workaround for CuDNN issue, install nvrtc.so (Triggered internally at /opt/conda/conda-bld/pytorch_1685259540419/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:80.)
  return F.conv2d(input, weight, bias, self.stride,
/home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/diffusers/configuration_utils.py:135: FutureWarning: Accessing config attribute `num_train_timesteps` directly via 'DDPMScheduler' object attribute is deprecated. Please access 'num_train_timesteps' over 'DDPMScheduler's config object instead, e.g. 'scheduler.config.num_train_timesteps'.
  deprecate("direct config name access", "1.0.0", deprecation_message, standard_warn=False)
/home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/torch/utils/checkpoint.py:391: UserWarning: torch.utils.checkpoint: please pass in use_reentrant=True or use_reentrant=False explicitly. The default value of use_reentrant will be updated to be False in the future. To maintain current behavior, pass use_reentrant=True. It is recommended that you use use_reentrant=False. Refer to docs for more details on the differences between the two variants.
  warnings.warn(
Steps:   4%|████████▋                                                                                                                                                                                                       | 625/15000 [05:50<2:17:46,  1.74it/s, lr=5e-5, step_loss=nan]05/29/2023 15:14:26 - INFO - __main__ - Running validation... 
 Generating 4 images with prompt: Generate a cartoonized version of the natural image.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
{'scaling_factor'} was not found in config. Values will be initialized to default values.
/home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/torch/utils/checkpoint.py:56: UserWarning: None of the inputs have requires_grad=True. Gradients will be None
  warnings.warn(
/home/kd/anaconda3/envs/sd/lib/python3.11/site-packages/diffusers/utils/pil_utils.py:38: RuntimeWarning: invalid value encountered in cast
  images = (images * 255).round().astype("uint8")
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Steps:   8%|█████████████████▎                                                                                                                                                                                             | 1250/15000 [11:48<2:11:02,  1.75it/s, lr=5e-5, step_loss=nan]05/29/2023 15:20:23 - INFO - __main__ - Running validation... 
 Generating 4 images with prompt: Generate a cartoonized version of the natural image.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
{'scaling_factor'} was not found in config. Values will be initialized to default values.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Steps:  12%|█████████████████████████▉                                                                                                                                                                                     | 1875/15000 [17:40<1:58:32,  1.85it/s, lr=5e-5, step_loss=nan]05/29/2023 15:26:15 - INFO - __main__ - Running validation... 
 Generating 4 images with prompt: Generate a cartoonized version of the natural image.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
{'scaling_factor'} was not found in config. Values will be initialized to default values.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Steps:  17%|██████████████████████████████████▌                                                                                                                                                                            | 2500/15000 [23:35<1:59:17,  1.75it/s, lr=5e-5, step_loss=nan]05/29/2023 15:32:10 - INFO - __main__ - Running validation... 
 Generating 4 images with prompt: Generate a cartoonized version of the natural image.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
{'scaling_factor'} was not found in config. Values will be initialized to default values.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Potential NSFW content was detected in one or more images. A black image will be returned instead. Try again with a different prompt and/or seed.
Steps:  21%|███████████████████████████████████████████▏                                                                                                                                                                   | 3125/15000 [29:41<1:50:02,  1.80it/s, lr=5e-5, step_loss=nan]05/29/2023 15:38:16 - INFO - __main__ - Running validation... 

```

**Comment by sayakpaul** [comment]
It might be because of the 8-bit Adam. Maybe try different learning rates and schedules to circumvent the issue. Unfortunately, we won't have cycles to reproduce this issue with the 8-bit settings.

**Comment by sayakpaul** [comment]
Closing this because of the above comment.
